File: 11.01/황지원/보물.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.StringTokenizer;

public class Boj_1026_보물 {
    static int n, a[];
    static Integer b[];

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st, st2;

        n = Integer.parseInt(br.readLine());
        a = new int[n];
        b = new Integer[n];

        st = new StringTokenizer(br.readLine());
        st2 = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(st.nextToken());
            b[i] = Integer.parseInt(st2.nextToken());
        }

        Arrays.sort(a);
        Arrays.sort(b, Collections.reverseOrder());

        long answer = 0;
        for(int i=0; i<n; i++){
            answer += a[i] * b[i];
        }
        System.out.println(answer);
    }
}

File: 2022/10.31/조수연/보물.kt
package com.ssafy.algorithm

import java.io.*
import java.util.*
import kotlin.collections.ArrayList


fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))

    val N = br.readLine().toInt()

    val A = ArrayList<Int>()
    val B = ArrayList<Int>()

    repeat(2) { time ->
        val token = StringTokenizer(br.readLine())

        repeat(N) {
            when (time) {
                0 -> A.add(token.nextToken().toInt())
                1 -> B.add(token.nextToken().toInt())
            }
        }
    }

    var sum = 0

    //A의 가장 큰수 * B의 가장 작은 수 곱하기 및 체크배열로 체크해주기
    while (A.isNotEmpty()){
        sum += A.maxOrNull()!! * B.minOrNull()!!
        A.remove(A.maxOrNull())
        B.remove(B.minOrNull())
    }

    println(sum)
}




